Typescript errors on import

gdax-tt v0.1.16
node v8.3.0
typescript 2.5.3

Trying to follow along with the getting started guide. Trying to compile file with GTT import leads to following failure

```typescript
import * as GTT from "gdax-trading-toolkit";
...
```

```
node_modules/gdax-trading-toolkit/build/src/core/HFTFilter.d.ts(22,22): error TS2415: Class 'HFTFilter' incorrectly extends base class 'Duplex'.
  Property '_write' is protected in type 'HFTFilter' but public in type 'Duplex'.
node_modules/gdax-trading-toolkit/build/src/core/LiveOrderbook.d.ts(39,22): error TS2415: Class 'LiveOrderbook' incorrectly extends base class 'Duplex'.
  Property '_write' is protected in type 'LiveOrderbook' but public in type 'Duplex'.
node_modules/gdax-trading-toolkit/build/src/core/MessageQueue.d.ts(67,22): error TS2415: Class 'MessageQueue' incorrectly extends base class 'Duplex'.
  Property '_write' is protected in type 'MessageQueue' but public in type 'Duplex'.
node_modules/gdax-trading-toolkit/build/src/core/RateLimiter.d.ts(21,22): error TS2415: Class 'RateLimiter' incorrectly extends base class 'Transform'.
  Property '_transform' is protected in type 'RateLimiter' but public in type 'Transform'.
node_modules/gdax-trading-toolkit/build/src/core/Trader.d.ts(46,22): error TS2415: Class 'Trader' incorrectly extends base class 'Writable'.
  Property '_write' is protected in type 'Trader' but public in type 'Writable'.
node_modules/gdax-trading-toolkit/build/src/exchanges/ExchangeFeed.d.ts(28,31): error TS2415: Class 'ExchangeFeed' incorrectly extends base class 'Readable'.
  Property '_read' is protected in type 'ExchangeFeed' but public in type 'Readable'.
node_modules/gdax-trading-toolkit/build/src/lib/StaticCommandSet.d.ts(20,22): error TS2415: Class 'StaticCommandSet' incorrectly extends base class 'Readable'.
  Property '_read' is protected in type 'StaticCommandSet' but public in type 'Readable'.
node_modules/gdax-trading-toolkit/build/src/utils/index.d.ts(3,29): error TS7016: Could not find a declaration file for module 'pushbullet'. '/app/node_modules/pushbullet/index.js' implicitly has an 'any' type.
```

--- a/src/core/HFTFilter.ts
+++ b/src/core/HFTFilter.ts
@@ -157,7 +157,7 @@ export class HFTFilter extends Duplex {
         });
     }
 
-    protected _read() {
+    _read() {
         /* The rules regarding readable streams are:
          - You have to call `push` to keep the stream alive.
          - Pushing 'null' ends the stream
@@ -176,7 +176,7 @@ export class HFTFilter extends Duplex {
     /**
      * If this stream has another stream piped into it, we just pass the data into the READABLE stream's filter
      */
-    protected _write(chunk: any, encoding: string, callback: (err: Error) => void) {
+    _write(chunk: any, encoding: string, callback: (err: Error) => void) {
         if (typeof chunk === 'object' && isStreamMessage(chunk)) {
             this.filterMessage(chunk as StreamMessage);
             return callback(null);

--- a/src/core/LiveOrderbook.ts
+++ b/src/core/LiveOrderbook.ts
@@ -149,9 +149,9 @@ export class LiveOrderbook extends Duplex implements Orderbook {
         return this._book.ordersForValue(side, Big(value), useQuote, startPrice);
     }
 
-    protected _read() { /* no-op */ }
+    _read() { /* no-op */ }
 
-    protected _write(msg: any, encoding: string, callback: () => void): void {
+    _write(msg: any, encoding: string, callback: () => void): void {
         // Pass the msg on to downstream users
         this.push(msg);
         // Process the message for the orderbook state

--- a/src/core/MessageQueue.ts
+++ b/src/core/MessageQueue.ts
@@ -135,11 +135,20 @@ export class MessageQueue extends Duplex {
         });
     }
 
+    _write(inputMessage: any, encoding: string, callback: (err: Error) => void): void {
+        if (this.defaultMessageHandler(inputMessage)) {
+            setImmediate(() => {
+                this._read();
+            });
+        }
+        callback(null);
+    }
+
     /**
      * Will provide the next message, in the correct order
      * @private
      */
-    protected _read() {
+    _read() {
         /* The rules regarding readable streams are:
          - You have to call `push` to keep the stream alive.
          - Pushing 'null' ends the stream
@@ -198,15 +207,6 @@ export class MessageQueue extends Duplex {
             }
         }
         return node;
-    }
-
-    protected _write(inputMessage: any, encoding: string, callback: (err: Error) => void): void {
-        if (this.defaultMessageHandler(inputMessage)) {
-            setImmediate(() => {
-                this._read();
-            });
-        }
-        callback(null);
     }
 
     private defaultMessageHandler(msg: any): boolean {

--- a/src/core/RateLimiter.ts
+++ b/src/core/RateLimiter.ts
@@ -33,7 +33,7 @@ export default class RateLimiter extends Transform {
         this.limiter = new Limiter(limit, interval, false);
     }
 
-    protected _transform(msg: StreamMessage, encoding: string, callback: (err?: Error, data?: any) => void): void {
+    _transform(msg: StreamMessage, encoding: string, callback: (err?: Error, data?: any) => void): void {
         this.limiter.removeTokens(1, () => {
             msg.time = new Date();
             this.push(msg);

--- a/src/core/Trader.ts
+++ b/src/core/Trader.ts
@@ -154,7 +154,7 @@ export class Trader extends Writable {
         });
     }
 
-    public executeMessage(msg: StreamMessage) {
+    executeMessage(msg: StreamMessage) {
         if (!isStreamMessage(msg)) {
             return;
         }
@@ -183,7 +183,7 @@ export class Trader extends Writable {
         }
     }
 
-    protected _write(msg: any, encoding: string, callback: (err?: Error) => any): void {
+    _write(msg: any, encoding: string, callback: (err?: Error) => any): void {
         this.executeMessage(msg);
         callback();
     }

--- a/src/exchanges/ExchangeFeed.ts
+++ b/src/exchanges/ExchangeFeed.ts
@@ -87,6 +87,11 @@ export abstract class ExchangeFeed extends Readable {
         this.close();
     }
 
+    _read(size: number) {
+        // This is not an on-demand service. For that, I refer you to Netflix. Data gets pushed to the queue as it comes
+        // in from the websocket, so there's nothing to do here.
+    }
+
     protected connect() {
         if (this._isConnecting || this.isConnected()) {
             return;
@@ -191,11 +196,6 @@ export abstract class ExchangeFeed extends Readable {
             this.log('error', 'Could not send message to GDAX WS server because the message was invalid',
                 { error: err, message: msg });
         }
-    }
-
-    protected _read(size: number) {
-        // This is not an on-demand service. For that, I refer you to Netflix. Data gets pushed to the queue as it comes
-        // in from the websocket, so there's nothing to do here.
     }
 }
 

--- a/src/lib/StaticCommandSet.ts
+++ b/src/lib/StaticCommandSet.ts
@@ -44,5 +44,5 @@ export class StaticCommandSet extends Readable {
         }
     }
 
-    protected _read(size: number): void { /* no-op */ }
+    _read(size: number): void { /* no-op */ }
 }

